source/common/http/conn_manager_impl.cc

```diff
@@ -598,7 +598,8 @@ void ConnectionManagerImpl::onEvent(Network::ConnectionEvent event) {
     // NOTE: In the case where a local close comes from outside the filter, this will cause any
     // stream closures to increment remote close stats. We should do better here in the future,
     // via the pre-close callback mentioned above.
-    doConnectionClose(absl::nullopt, absl::nullopt, details);
+    doConnectionClose(absl::nullopt, StreamInfo::CoreResponseFlag::DownstreamConnectionTermination,
```

Comment:
should we then remove the now redundant setting of  remote_close in ConnectionManagerImpl::ActiveStream::completeRequest() { ?  Not sure if it's safe to move the response code details setting inline but it seems plausible we could clean both up.

Also can you add a comment at the definition of DownstreamConnectionTermination to disambiguate it's (now) the termination of the downstream connection not a connection termination initiated by downstream?

Reply:
Thanks for the quick review! 

Good point, I think it is safe and removed it. The comment right now is `// If the stream was reset due to a downstream connection termination.`, which looks reasonable to me.